src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombiner.java

```diff
@@ -131,13 +132,18 @@ private void updateQueryTopDocsWithCombinedScores(
         compoundQueryTopDocs.setTotalHits(getTotalHits(topDocsPerSubQuery, maxHits));
     }
 
+    /**
+     * Get max hits as number of unique doc ids from results of all sub-queries
+     * @param topDocsPerSubQuery list of topDocs objects for one shard
+     * @return number of unique doc ids
+     */
     protected int getMaxHits(final List<TopDocs> topDocsPerSubQuery) {
-        int maxHits = 0;
-        for (TopDocs topDocs : topDocsPerSubQuery) {
-            int hits = topDocs.scoreDocs.length;
-            maxHits = Math.max(maxHits, hits);
-        }
-        return maxHits;
+        Set<Integer> docIds = topDocsPerSubQuery.stream()
+            .filter(topDocs -> Objects.nonNull(topDocs.scoreDocs))
+            .flatMap(topDocs -> Arrays.stream(topDocs.scoreDocs))
+            .map(scoreDoc -> scoreDoc.doc)
+            .collect(Collectors.toSet());
+        return docIds.size();
```

Comment:
will this work in the case when no valid documents are found from both the queries and we are adding headers and footers in score docs?

Reply:
Yes, this scenario is handled properly. Parsing of the protocol for hybrid query results happens at earlier stage of workflow: in [workflow class we're constructing list of CompoundTopDocs objects per shard](https://github.com/martin-gaievski/neural-search/blob/aggregations_in_hybrid_query/src/main/java/org/opensearch/neuralsearch/processor/NormalizationProcessorWorkflow.java#L59) and scores are disassembled into individual results per sub-query in [CompoundDocs constructor](https://github.com/martin-gaievski/neural-search/blob/aggregations_in_hybrid_query/src/main/java/org/opensearch/neuralsearch/processor/CompoundTopDocs.java#L76-L98).